### PR TITLE
test: Add initial e2e test

### PR DESCRIPTION
**Pull request by animeshk08**
This commit add e2e test to create a
volume on demand with mount options.

Signed-off-by: Animesh Kumar <animuz111@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test

> /kind flake

/kind feature

**What this PR does / why we need it**:
Adds intial implementation of e2e tests
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
test: Add initial e2e test
```

### CODE REVIEW

**Comment by animeshk08** [comment]
Initial implementation of the e2e test. The test ran successfully on a minikube cluster after manually deploying the driver and samba server.

Since the target is to test it out on minikube the boilerplate code related to azure connection establishment(azure credentials, clients etc.) has been removed for now.

**Comment by animeshk08** [comment]
Partially fixes: #9

**Comment by andyzhangx** [comment]
@animeshk08 thanks, if it works, then we could set up the e2e test in travis CI. Is this PR ready to merge?

**Comment by animeshk08** [comment]
@andyzhangx To run the e2e test on travis I should install and create a kubernetes cluster on travis right?

Also, I noticed one thing. Inside the make file the name of the helm repository(`smb-csi-driver`) is not the same as that present in the charts(I named it `csi-driver-smb` after the name of the github repo itself). Let me know which one you want to retain, and I can send a PR for the fix. Ref: https://github.com/kubernetes-csi/csi-driver-smb/blob/1ca7c4879892831bcb15e8751069d31ba9678c93/Makefile#L72

**Comment by animeshk08** [comment]
@andyzhangx The PR is ready for review. Please have a look at the integration test logs also here: https://travis-ci.org/github/kubernetes-csi/csi-driver-smb/jobs/711681088

File: .travis.yml [hunk 1]
```diff
@@ -32,3 +30,35 @@ jobs:
...
+        - sudo apt-get -qq -y install conntrack
+      env:
+        - CHANGE_MINIKUBE_NONE_USER=true
+        - MINIKUBE_WANTUPDATENOTIFICATION=false
+        - MINIKUBE_WANTREPORTERRORPROMPT=false
+        - MINIKUBE_HOME=$HOME
+        - CHANGE_MINIKUBE_NONE_USER=true
+        - KUBECONFIG=$HOME/.kube/config
+      before_script:
+        - curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.18.1/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
```
Comment (andyzhangx) [inline .travis.yml:45]: is it possible to put these commands into one script file? so others could also set up minikube cluster in local env?

Comment (animeshk08) [reply .travis.yml]: Hi @andyzhangx I have made the changes. The script is present in `test/utils`. Thanks

**Comment by andyzhangx** [comment]
@animeshk08 do you know how to fix this e2e test error in the last:
```
 [SLOW TEST:15.335 seconds]
Dynamic Provisioning
/home/travis/gopath/src/github.com/csi-driver/csi-driver-smb/test/e2e/dynamic_provisioning_test.go:34
  should create a volume on demand with mount options [kubernetes.io/smb-csi] [smb.csi.k8s.io] [Windows]
  /home/travis/gopath/src/github.com/csi-driver/csi-driver-smb/test/e2e/dynamic_provisioning_test.go:63
------------------------------
Failed to create JUnit directory: /workspace/_artifacts/junit_01.xml
	mkdir /workspace: permission deniedFailed to create JUnit report file: /workspace/_artifacts/junit_01.xml
	open /workspace/_artifacts/junit_01.xml: no such file or directory
Failed to generate JUnit report data:
	invalid argument
```

File: test/utils/deploy-minikube.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+minikube update-context --profile=minikube
+sudo chown -R travis: /home/travis/.minikube/
+
+# setup samba server and deploy SMB CSI driver
+kubectl cluster-info
+JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until kubectl -n kube-system get pods -lk8s-app=kube-dns -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 1;echo "waiting for kube-dns to be available";  done
+kubectl create secret generic smbcreds --from-literal username=USERNAME --from-literal password="PASSWORD"
+kubectl create -f deploy/example/smb-provisioner/smb-server.yaml
+JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'; until kubectl -n default get pods -lapp=smb-server -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 1;echo "waiting for smb controller deployment to be available"; done
+bash deploy/install-driver.sh
```
Comment (andyzhangx) [inline test/utils/deploy-minikube.sh:38]: here the e2e test still installs official latest tag of driver image, I think you should use helm installation to install current driver: https://github.com/kubernetes-csi/csi-driver-smb/blob/1ca7c4879892831bcb15e8751069d31ba9678c93/Makefile#L68-L83

**Comment by andyzhangx** [comment]
wait for this PR first: https://github.com/kubernetes-csi/csi-driver-smb/pull/72

**Comment by animeshk08** [comment]
@andyzhangx I have fixed the error. The PR is now ready to be merged. Thank you for your help!!

**Pull request merged**